components/director/internal/tenantfetchersvc/resync/tenant_manager.go

```diff
@@ -336,27 +339,109 @@ func walkThroughPages(ctx context.Context, eventAPIClient EventAPIClient, events
 		return err
 	}
 
-	for i := pageStart + 1; i <= totalPages; i++ {
-		params[pageConfig.PageNumField] = strconv.FormatInt(i, 10)
-		res, err := eventAPIClient.FetchTenantEventsPage(ctx, eventsType, params)
-		if err != nil {
-			return errors.Wrap(err, "while fetching tenant events page")
-		}
-		if res == nil {
-			return apperrors.NewInternalError("next page was expected but response was empty")
-		}
-		if initialCount != gjson.GetBytes(res.Payload, pageConfig.TotalResultsField).Int() {
-			return apperrors.NewInternalError("total results number changed during fetching consecutive events pages")
-		}
+	region := getRegionFromCtx(ctx)
+	log.C(ctx).Infof("Starting processing for %d events across %d pages for event type %v in region %s", initialCount, totalPages, eventsType, region)
+
+	start := time.Now()
+	var m sync.Mutex
+	eventPages := make([]*EventsPage, 0)
+	pagesQueue := make(chan QueryParams)
```

Comment:
`pagesQueue` is kind of confusing to me and misleading as it doesn't hold any pages. It's longer, but `pagesQueryParamsQueue`/`pagesQParamsQueue`/`queryParamsQueue` is more correct here.